src/inmanta/ast/constraint/expression.py

```diff
@@ -521,7 +521,7 @@ def _bin_op(self, arg1: object, arg2: object) -> object:
         return self._arithmetic_op(arg1, arg2)
 
     @abstractmethod
-    def _arithmetic_op(self, arg1: object, arg2: object) -> object:
+    def _arithmetic_op(self, arg1: numbers.Number, arg2: numbers.Number) -> object:
```

Comment:
Why not using this same signature for the Subclasses as well ?

Reply:
I changed all types to `int|float`, because mypy was complaining. The type `number.Number` is too wide.